src/AppInstallerCLICore/ExecutionContext.cpp

```diff
@@ -192,19 +192,19 @@ namespace AppInstaller::CLI::Execution
         Reporter.CancelInProgressTask(bypassUser);
     }
 
-    void Context::SetExecutionStage(Workflow::ExecutionStage stage, bool allowBackward)
+    void Context::SetExecutionStage(Workflow::ExecutionStage stage)
     {
         if (m_executionStage == stage)
         {
             return;
         }
-        else if (m_executionStage > stage && !allowBackward)
+        else if (m_executionStage > stage)
         {
             THROW_HR_MSG(HRESULT_FROM_WIN32(ERROR_INVALID_STATE), "Reporting ExecutionStage to an earlier Stage without allowBackward as true");
         }
 
         m_executionStage = stage;
-        Logging::SetExecutionStage(static_cast<uint32_t>(m_executionStage));
+        Logging::Telemetry().SetExecutionStage(static_cast<uint32_t>(m_executionStage));

```

Comment:
This seems like madness.  The owner of the thread specific telemetry object requesting it via the thread locals rather than getting it directly.